src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Sign.cs

```diff
@@ -30,20 +30,20 @@ public static void Sign<T>(ReadOnlySpan<T> x, Span<int> destination)
         {
             public static unsafe bool Vectorizable =>
                 // TODO: Extend vectorization to handle primitives whose size is not the same as int
-                typeof(T) == typeof(uint) || typeof(T) == typeof(int) || typeof(T) == typeof(float);
+                IsInt32Like<T>() || IsUInt32Like<T>() || typeof(T) == typeof(float);
```

Comment:
This could be `Vector128<T>.IsSupported && (sizeof(T) == sizeof(int))`, along the general lines of what was said above.

This whole method could really just easily support anything that `Vector128<T>.IsSupported` returns true for. We can just use `IsNaN`, `IsNegative`, and `IsZero` to handle everything. The JIT already constant folds these to dead code for `T` where they can't be true.

Reply:
> sizeof(T) == sizeof(int)

Done

> This whole method could really just easily support anything that Vector128<T>.IsSupported returns true for.

I'm not understanding. The restriction to `sizeof(int)` is because it's producing `VectorXx<int>` output vectors, so different sized inputs would require much different handling due to the different number of elements between input and output.

Reply:
> I'm not understanding. The restriction to sizeof(int) is because it's producing VectorXx<int> output vectors, so different sized inputs would require much different handling due to the different number of elements between input and output.

Ah, right. I forgot this is explicitly returning `int` because its one of the very old .NET Framework APIs. We generally want modern code to use `CopySign` or one of the `Is*` queries instead.

Nevermind then.